pkg/bundle/bundle.go

```diff
@@ -238,7 +248,17 @@ func (b *bundle) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result,
 
 	log.V(2).Info("successfully synced bundle")
 
-	b.setBundleCondition(&bundle, syncedCondition)
+	b.setBundleCondition(
+		bundle.Status.Conditions,
+		&bundle.Status.Conditions,
+		trustapi.BundleCondition{
+			Type:               syncedCondition.Type,
+			Status:             syncedCondition.Status,
+			Reason:             syncedCondition.Reason,
+			Message:            syncedCondition.Message,
+			ObservedGeneration: bundle.Generation,
```

Comment:
Could it make sense to set `ObservedGeneration` when creating `syncedCondition` - to simplify the code? Or is there a reason for doing it like this?

Reply:
I also refactored bundleHasCondition and moved the ObservedGeneration to syncedCondition